### ISSUE TITLE
Config is ignored when opening file (if not loaded in time)

### ISSUE THREAD

**Issue by andys8** [fannheyward/coc-markdownlint#65]
Hi, looking forward to use `coc-markdownlint`.

I'm giving it a try and I wanted to disable line length warnings. I added this to `coc-settings.json` (as described in `README`).

```
  "markdownlint.config": {
    "default": true,
    "line_length": false
  },
```

It looks like it has no effect.

![image](https://user-images.githubusercontent.com/13085980/71832309-f8810500-30aa-11ea-91c6-5ba7ad03f76c.png)

### `:CocInfo`

```
## Output channel: markdownlint

[5:36:07 PM] Info: global config: {"_":[]}
[5:36:07 PM] Info: config from coc-settings.json: {"default":true,"line_length":false}
[5:36:07 PM] Info: full config: {"_":[],"default":true,"line_length":false}
```

Since the plugin is quite new, I thought it's a good idea to create an issue and report it. Maybe I'm doing something wrong.

**Comment by andys8** [comment]
In comparison: A local file `.markdownlint.json` with the same content has an effect.

**Comment by andys8** [comment]
I think the issue is that config is being loaded. But linting is triggered and just grabs the current `config` independant if it's the default or actually loaded.

https://github.com/fannheyward/coc-markdownlint/blob/0f5efb5a49b9c26f38816338182bd8af81b40c63/src/engine.ts#L67

So a promise instead of the `config` object could do the trick.
